### PR TITLE
Support kubernetes 1.31

### PR DESCRIPTION
**Pull request by toshipp**
fix: https://github.com/topolvm/topolvm/issues/951

### CODE REVIEW

**Review by cupnes — COMMENTED** [review]
The definition of the `external-snapshotter-leaderelection` role can be found here:
https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/rbac-csi-snapshotter.yaml

So, please also update the following comment:
https://github.com/topolvm/topolvm/blob/90c0e6f3c54d2e7d5027c2316344eb0fa8436163/charts/topolvm/templates/controller/roles.yaml#L59

File: .golangci.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ linters:
     # The rules violated to make lint work are commented out. We will fix it eventually.
     - dupl
     - errcheck
-    - exportloopref
+    - copyloopvar
```
Comment (llamerada-jp) [inline .golangci.yml:29]: I couldn't find the reason for this change. Could you tell me why this change was caused?

File: go.mod [hunk 2]
```diff
@@ -3,68 +3,73 @@ module github.com/topolvm/topolvm
 go 1.22.0
```
[the remark this answers is not in the document]
Comment (toshipp) [reply go.mod:3]: The go directive does not specify the go command version. We refer to the version in kubernetes go.mod, not go command version.
https://github.com/kubernetes/kubernetes/blob/release-1.31/go.mod#L9

File: .golangci.yml (hunk 1 above)
Comment (toshipp) [reply .golangci.yml:29]: Please see the commit message. https://github.com/topolvm/topolvm/pull/978/commits/368fd9c7d73dde01cc26a84bf1f2686e36bce04c
golintci-lint ouputs the deprecation message.

File: versions.mk [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-CERT_MANAGER_VERSION := v1.15.1
+CERT_MANAGER_VERSION := v1.16.1
 # https://github.com/helm/chart-testing/releases
 CHART_TESTING_VERSION := 3.11.0
 # https://github.com/containernetworking/plugins/releases
-CNI_PLUGINS_VERSION := v1.5.1
+CNI_PLUGINS_VERSION := v1.6.0
 # https://github.com/GoogleContainerTools/container-structure-test/releases
-CONTAINER_STRUCTURE_TEST_VERSION := 1.18.1
+CONTAINER_STRUCTURE_TEST_VERSION := 1.19.1
```
[the remark this answers is not in the document]
Comment (toshipp) [reply versions.mk:11]: I found two new versions were released in one day, it is too high speed. I want to use the current version to avoid another update while reviewing.

File: versions.mk [hunk 4]
```diff
@@ -1,34 +1,34 @@
...
+KIND_VERSION := v0.24.0
 # It is set by CI using the environment variable, use conditional assignment.
-KUBERNETES_VERSION ?= 1.30.0
+KUBERNETES_VERSION ?= 1.31.0
 # https://github.com/kubernetes/minikube/releases
-MINIKUBE_VERSION := v1.33.1
+MINIKUBE_VERSION := v1.34.0
 # https://github.com/protocolbuffers/protobuf/releases
-PROTOC_VERSION :=  27.2
+PROTOC_VERSION := 28.2
```
[the remark this answers is not in the document]
Comment (toshipp) [reply versions.mk:31]: Oh, it was released just 10 hours ago. Same as above. I want to keep the current version.

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (cupnes) [reply go.mod:3]: Thank you. I understood.

**Review by cupnes — APPROVED** [review]

**Review by llamerada-jp — APPROVED** [review]
